cmd/thanos/sidecar.go

```diff
@@ -298,13 +320,45 @@ func runSidecar(
 	return nil
 }
 
+func validatePrometheus(ctx context.Context, logger log.Logger, m *promMetadata) error {
+	if m.version == nil {
+		level.Warn(logger).Log("msg", "fetched version is nil or invalid. Unable to know whether Prometheus supports /version endpoint, skip validation")
+		return nil
+	}
+
+	if m.version.LessThan(promclient.FlagsVersion) {
+		level.Warn(logger).Log("msg",
+			"Prometheus doesn't support flags endpoint, skip validation", "version", m.version.Original())
+		return nil
+	}
+
+	flags, err := promclient.ConfiguredFlags(ctx, logger, m.promURL)
+	if err != nil {
+		return errors.Wrap(err, "failed to check flags")
+	}
+
+	// Check if compaction is disabled.
+	if flags.TSDBMinTime != flags.TSDBMaxTime {
+		return errors.Errorf("Found that TSDB Max time is %s and Min time is %s. "+
```

Comment:
Please lower the case of the first letter here too

Reply:
Done.